src/compiler/factory.ts

```diff
@@ -1440,6 +1440,28 @@ namespace ts {
             : node;
     }
 
+    export function createInterfaceDeclaration(decorators: Decorator[] | undefined, modifiers: Modifier[] | undefined, name: string | Identifier | undefined, typeParameters: TypeParameterDeclaration[] | undefined, heritageClauses: HeritageClause[], members: TypeElement[]) {

```

Comment:
`heritageClauses` can be `undefined`, so the type should be `HeritageClause[] | undefined`.